node/middlewares/customRoutes.ts

```diff
@@ -0,0 +1,52 @@
+import { getAppsRoutes, getUserRoutes } from '../services/routes'
+
+export async function customRoutes(ctx: Context, next: () => Promise<void>) {
+  const start = process.hrtime()
+
+  const {
+    vtex: { logger },
+  } = ctx
+
+  const fetchRoutes = async (
+    fetchFunction: (ctx: Context) => Promise<string[]>,
+    metricName: string
+  ) =>
+    fetchFunction(ctx).catch(err => {
+      const diff = process.hrtime(start)
+      metrics.batch(metricName, diff)
+      logger.error({
+        error: err,
+        type: metricName,
+      })
+      return null
+    })
```

Comment:
```suggestion
const fetchRoutesWithMetrics = async (
    fetchFunction: (ctx: Context) => Promise<string[]>,
    metricConfig: { failed: string, success?: string }
  ) => {
    try {
      return await fetchFunction(ctx);
    } catch (error) {
      const timeDiff = process.hrtime(startTime);
      metrics.batch(metricConfig.failed, timeDiff);
      logger.error({ error, type: 'error' });
      return null;
    }
  };
```
Should we also add metrics for success ?

Reply:
Applied with the addition of the log for redirect succes